part3-mission4/src/routes/user-router.js

```diff
@@ -0,0 +1,96 @@
+import express from 'express';
+import { userController } from '../controllers/user-controller.js';
+import { productController } from '../controllers/product-controller.js';
+import { validation } from '../middlewares/validation.js';
+import passport from '../lib/passport/index.js';
+import authenticate from '../middlewares/authenticate.js';
+import { isUserSelf } from '../middlewares/authorize.js';
+import { articleController } from '../controllers/article-controller.js';
+
+const router = express.Router();
+
+// 등록
+router.post('/register', validation.validateUsername, userController.register);
+
+// 로그인&로그아웃
+router.post(
+  '/login',
+  passport.authenticate('local', { session: false }),
+  userController.login
+);
+router.post('/logout', userController.logout);
+
+// 유저 조회, 정보 수정
+router
+  .route('/:userId')
+  .get(
+    authenticate,
+    isUserSelf,
+    validation.validateParam('id', validation.idSchema),
```

Comment:
적절한 변수명으로 수정하는 것이 바람직할 것 같습니다.

```suggestion
    validation.validateParam('userId', validation.idSchema),
```

Reply:
파일 전반적으로 확인해주세요!